src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -384,68 +463,34 @@ private[table] class IndexedTableImpl(
-
-    val currentIndexStatus = snapshot.loadIndexStatus(revisionID)
-    val cubesToOptimize = bo.cubesToOptimize.map(currentIndexStatus.revision.createCubeId)
-    val indexStatus = currentIndexStatus.addAnnouncements(cubesToOptimize)
-    val cubesToReplicate = indexStatus.cubesToOptimize
+  override def optimize(files: Seq[String]): Unit = {
+    val paths = files.toSet
     val schema = metadataManager.loadCurrentSchema(tableID)
-
-    if (cubesToReplicate.nonEmpty) {
-      try {
-        // Try to commit transaction
-        doOptimize(schema, indexStatus, cubesToReplicate)
-      } catch {
-        case _: ConcurrentModificationException => bo.end(Set())
-      } finally {
-        // end keeper transaction
-        bo.end(cubesToReplicate.map(_.string))
+    snapshot.loadAllRevisions.foreach { revision =>
+      val indexFiles = snapshot
+        .loadIndexFiles(revision.revisionID)
+        .filter(file => paths.contains(file.path))
+        .toIndexedSeq
+      if (indexFiles.nonEmpty) {
+        val indexStatus = snapshot.loadIndexStatus(revision.revisionID)
+        metadataManager.updateWithTransaction(
+          tableID,
+          schema,
+          QbeastOptions.empty,
+          append = true) {
+          val tableChanges = BroadcastedTableChanges(None, indexStatus, Map.empty, Map.empty)
+          val fileActions =
+            dataWriter.compact(tableID, schema, revision, indexStatus, indexFiles)
+          (tableChanges, fileActions)
+        }
       }
-    } else {
-      bo.end(Set())
-    }
-
-    clearCaches()
-  }
-
-  private def doOptimize(
-      schema: StructType,
-      indexStatus: IndexStatus,
-      cubesToOptimize: Set[CubeId]): Unit = {
-
-    metadataManager.updateWithTransaction(tableID, schema, QbeastOptions.empty, append = true) {
-      val dataToReplicate =
-        CubeDataLoader(tableID).loadSetWithCubeColumn(
-          cubesToOptimize,
-          indexStatus.revision,
-          QbeastColumns.cubeToReplicateColumnName)
-      val (qbeastData, tableChanges) =
-        indexManager.optimize(dataToReplicate, indexStatus)
-      val fileActions =
-        dataWriter.write(tableID, schema, qbeastData, tableChanges)
-      (tableChanges, fileActions)
     }
-
   }
 
-  override def compact(revisionID: RevisionID): Unit = {
-
-    // Load the schema and the current status
-    val schema = metadataManager.loadCurrentSchema(tableID)
-    val currentIndexStatus = snapshot.loadIndexStatus(revisionID)
-
-    metadataManager.updateWithTransaction(tableID, schema, QbeastOptions.empty, append = true) {
-      // There's no affected table changes on compaction, so we send an empty object
-      val tableChanges = BroadcastedTableChanges(None, currentIndexStatus, Map.empty)
-      val fileActions =
-        dataWriter.compact(tableID, schema, currentIndexStatus, tableChanges)
-      (tableChanges, fileActions)
-
-    }
-
-  }
+  override def compact(revisionID: RevisionID): Unit = optimize(revisionID)
```

Comment:
Do we still need this command?

Reply:
I guess we don't since the only purpose it's calling `optimize`.